### DISCUSSION TITLE
Password problem with seq in docker compose

### DISCUSSION THREAD

**Discussion by scs-tec** [datalust/seq-tickets#1923] [Help]
Hey, 

i am trying out seq with docker-compose on windows. The container starts and i can access the webpage, but i can't login.

from console: 
`echo 'MyStrongPassword123!' | docker run --rm -i datalust/seq config hash
QBpgwsoe5E1ePbrSX80nqYhJ9Sl4YE+8SzyINj7IX1Dmh1kkCzJyj6cH+9LIacqSMcC0ZBmRRTya6ZFfIp5GJOrgvJXAQTZoIKSmY7rADvET`

and here is the container in my docker-compose.yaml

`seq:
     image: datalust/seq
     ports:
      - 5341:5341
      - 80:80
     environment:
       - SEQ_FIRSTRUN_ADMINPASSWORDHASH=QBpgwsoe5E1ePbrSX80nqYhJ9Sl4YE+8SzyINj7IX1Dmh1kkCzJyj6cH+9LIacqSMcC0ZBmRRTya6ZFfIp5GJOrgvJXAQTZoIKSmY7rADvET
       - ACCEPT_EULA=Y
     volumes:
      - seq-appdata:/data`

anyone knows what the mistake is?

From Seq.json:

 "firstRun": {
    "adminPasswordHash": null,
    "adminUsername": null,
    "requireAuthenticationForHttpIngestion": false
  },

Kind regards

**Comment by KodrAus** [comment]
Hi @scs-tec :wave:

Thanks for reaching out. Some things you can try are:

1. Deleting your `seq-appdata` directory and letting it re-initialize.
2. Putting quotes around the `SEQ_FIRSTRUN_ADMINPASSWORDHASH` argument in case it's being interpreted by YAML weirdly.

If neither of those work, would it be possible to see your whole `docker-compose.yaml`?

**Comment by scs-tec** [comment]
environment:
       - "ACCEPT_EULA=Y"
       - "SEQ_FIRSTRUN_ADMINPASSWORDHASH=QAW7jJbw0eyOtzedBqBPKbKwFotTlNQzrI34oLLdgDrgsVTAf2MMNXf+l9wL7ufCrQw5zu9ByK+5JE8dnDLu2cVTMF5FSNsxPOmDir4IMLXw"

**Reply by KodrAus** [reply]
That YAML seems valid. Which options did you end up trying that didn't work @scs-tec?

**Reply by scs-tec** [reply]
I have recreated the hash multiple times. And I deleted the volumne, as you said. I can use the docker container without authentification, but if i set the admin password, the webpage stills say that the password is not correct. I am not sure about how to go along.

**Reply by nblumhardt** [reply]
Thanks for the follow-up. What happens if you just `docker run` the container, passing your generated hash on the command-line using `-e SEQ_FIRSTRUN_ADMINPASSWORDHASH=...`?

Trying it with no mounted volume at all would rule out some possible issues.

If you run this with `-it`, you might spot a more detailed failure message in its output.

It might be worth me mentioning that the `SEQ_FIRSTRUN_*` variables are only consulted when the data store is first initialized; they'll be ignored as soon as the container has started up successfully the first time over a given data store.

HTH!

**Comment by hookenful** [comment]
Bro it is still doesnt work. Tried to remove volume / start without volume / update hash/ start from docker run / start from docker-compose. Still no password data in Seq.json. Can you try that by yourself?

```
seq:
    image: datalust/seq:latest
    container_name: seq
    volumes:
     - seq:/data
    environment:
      - ACCEPT_EULA=Y
      - SEQ_FIRSTRUN_ADMINUSERNAME=admin
      - SEQ_FIRSTRUN_ADMINPASSWORDHASH=FCfNlV8nyAIBkrXHtKGK3A6rJ15CFPNykGLdEsi98tUeqFInZg==
      - SEQ_FIRSTRUN_REQUIREAUTHENTICATIONFORHTTPINGESTION=True
    ports:
      - 5341:5341
      - 8082:80
```

```
  "firstRun": {
    "adminPasswordHash": null,
    "adminPassword": null,
    "adminUsername": null,
    "requireAuthenticationForHttpIngestion": false
  },
```


```
PS C:\Users\hooki\Documents\coffeeface> docker run --rm -e ACCEPT_EULA=Y -e SEQ_FIRSTRUN_ADMINPASSWORDHASH=QBpgwsoe5E1ePbrSX80nqYhJ9Sl4YE+8SzyINj7IX1Dmh1kkCzJyj6cH+9LIacqSMcC0ZBmRRTya6ZFfIp5GJOrgvJXAQTZoIKSmY7rADvET -it  datalust/seq:latest
────────────────────────────────────────
  Seq ♦ Machine data, for humans.
─────────── © 2024 Datalust Pty Ltd ────

Running as server; press Ctrl+C to exit.

[12:26:28 INF] Seq 2024.3.11510 running on OS Ubuntu 22.04.4 LTS
[12:26:28 INF] Opening event store at /data/Stream/stream.flare
[12:26:28 INF] Ingestion enabled
[12:26:28 INF] Opening metastore /data/Documents/metastore.flare
[12:26:28 INF] Initializing a new metastore
[12:26:28 INF] Enabling username/password authentication, and using the supplied default admin password or hash
[12:26:28 INF] Creating default workspace for user-admin
[12:26:28 INF] Storage subsystem available
[12:26:28 INF] Seq listening on ["http://localhost/", "https://localhost/", "http://localhost:5341/", "https://localhost:45341/"]
```

**Reply by nblumhardt** [reply]
Using the plain text `SEQ_FIRSTRUN_ADMINPASSWORD` variable instead of the `*HASH` version might be an easier option, depending on your deployment requirements.

**Reply by mloskot** [reply]
@hookenful 
> Still no password data in Seq.json

Apologies for necromancy, but I've got confused myself about the similar behaviour. See my question in https://github.com/datalust/seq-tickets/discussions/2461

**Comment by schumacher-dev** [comment]
I was in the same issue. You can use SEQ_FIRSTRUN_ADMINPASSWORD informing plain-text password instead of SEQ_FIRSTRUN_ADMINPASSWORDHASH.  It works to me

**Comment by Sjaaky** [comment]
Using echo on windows is different than echo on linux.
`echo 'MyStrongPassword123!'` on windows echo's the quotes:
`'MyStrongPassword123!'`
while it doesn't on linux.
Furthermore echo on windows and linux use a different line ending. Depending on the implementation of seqs config hash function, that might be an issues as well. 

I was just searching how to get a password hash and found this issue first.

But even knowing this, it is still not working for me 😕 .

**Reply by mloskot** [reply]
@Sjaaky 
> But even knowing this, it is still not working for me 😕 .

What do you mean by the "it"?

If you cannot find the working command in the legacy Command Prompt, then try PowerShell.
Here is what I do in my PowerShell script in order to generate the password hash custom for custom deployment on Windows:

```powershell
$seqCmd = Get-Command -Name 'C:/Seq/Seq.exe' -CommandType Application -ErrorAction Stop
$env:SEQ_FIRSTRUN_ADMINPASSWORDHASH = $env:MY_SEQ_PASSWORD | & $seqCmd config hash
```

The important part is the `config hash` invocation, without any `echo`-ing.
